netci.groovy

```diff
@@ -1624,18 +1679,100 @@ def static addTriggers(def job, def branch, def isPR, def architecture, def os,
+    def needsTrigger = true
+    def isDefaultTrigger = false
+    def isArm64PrivateJob = false
+    def scenarioString = ""
+
+    // Set up default context string and trigger phrases. This is overridden in places, sometimes just to keep
+    // the existing non-standard descriptions and phrases. In some cases, the scenarios are asymmetric, as for
+    // some jobs where the Debug configuration just does builds, no tests.
+    //
+    // Some configurations, like arm32/arm64, always use the exact scenario name as part of the context string.
+    // This makes it possible to copy/paste the displayed context string as "@dotnet-bot test <context-string>"
+    // to invoke the trigger. Any "fancy" context string makes that impossible, requiring the user to either 
+    // remember the mapping from context string to trigger string, or use "@dotnet-bot help" to look it up.
+
+    if (architecture == 'armem') {
+        assert os == 'Tizen'
+        architecture = 'armel'
+    }
+
+    switch (architecture) {
+        case 'x64_arm64_altjit':
+        case 'x86_arm_altjit':
+            // TODO: for consistency, add "Build and Test" at end.
+            contextString = "${os} ${architecture} ${configuration} ${scenario}"
+            triggerString = "(?i).*test\\W+${os}\\W+${architecture}\\W+${configuration}\\W+${scenario}.*"
+            break
+
+        case 'armel':
+        case 'arm':
+        case 'arm64':
+            contextString = "${os} ${architecture} Cross ${configuration}"
+            triggerString = "(?i).*test\\W+${os}\\W+${architecture}\\W+Cross\\W+${configuration}"
+
+            if (scenario == 'innerloop') {
+                contextString += " Innerloop"
+                triggerString += "\\W+Innerloop"
+            }
+            else {
+                contextString += " ${scenario}"
+                triggerString += "\\W+${scenario}"
+            }
+
+            if (configuration == 'Debug') {
+                contextString += " Build"
+                triggerString += "\\W+Build"
+            }
+            else {
+                contextString += " Build and Test"
+                triggerString += "\\W+Build and Test"
+            }
+
+            triggerString += ".*"
+            break
+
+        default:
+            scenarioString = getScenarioDisplayString(scenario)
+            contextString = "${os} ${architecture} ${configuration} ${scenarioString}"
+            triggerString = "(?i).*test\\W+${os}\\W+${architecture}\\W+${configuration}"
+
+            switch (scenario) {
+                case 'normal':
+                    triggerString += "\\W+Build and Test.*"
+                    break
+
+                case 'corefx_innerloop': // maintain this asymmetry
+                    triggerString += "\\W+CoreFX Tests.*"
+                    break
+
+                default:
+                    triggerString += "\\W+${scenario}.*"
+                    break
+            }
+
+            triggerString += ".*"
```

Comment:
Do we need this extra `.*` in trigger string regex?

Reply:
It's always been there, so I didn't change it. Seems unlikely it's actually needed.